### PR TITLE
Torque layer reloading

### LINKED ISSUE #1266: calling update on a torque layer reload the tiles
When updating a torque layer using the `update` method, changing cartocss triggers a full map reload.

It also fails to update the style.

steps to reproduce:

1) run this code
```
layer.update("Map { -torque-frame-count: 256; -torque-animation-duration: 30; -torque-time-attribute: "cartodb_id"; -torque-aggregation-function: "count(1)"; -torque-resolution: 2; -torque-data-aggregation: linear; }#layer { marker-width: 10; marker-fill: #29a846; marker-fill-opacity: 1; marker-allow-overlap: true; marker-line-width: 2; marker-line-color: #FFFFFF; marker-line-opacity: 1; }");

layer.update("Map { -torque-frame-count: 256; -torque-animation-duration: 30; -torque-time-attribute: "cartodb_id"; -torque-aggregation-function: "count(1)"; -torque-resolution: 2; -torque-data-aggregation: linear; }#layer { marker-width: 10; marker-fill: #F00; marker-fill-opacity: 1; marker-allow-overlap: true; marker-line-width: 2; marker-line-color: #FFFFFF; marker-line-opacity: 1; }"); // change marker-fill
```

tiles are reloaded, the style is not update and it should.

Take into account if `torque-time-attribute` was the change property it should reload map tiles

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
(Hopefully) closes #1266.

@javisantana: yay or nay?

### CODE REVIEW

File: src/geo/map/torque-layer.js [hunk 1]
```diff
@@ -18,12 +18,36 @@ var TorqueLayer = LayerModelBase.extend({
...
+        return true;
+      }
+    }, this);
+
+    if (reloadMap) {
+      this._reloadMap();
+    }
+  },
+
+  _torqueTimeAttributeCartoCSSPropChanged: function () {
```
Comment (javisantana) [inline src/geo/map/torque-layer.js]: I don't think this is the right way to do it, check this method that allows to get properties from cartocss https://github.com/CartoDB/torque/blob/31a81e760b249e76b159c9966c219ae570f9acb3/lib/torque/common.js#L33

Comment (alonsogarciapablo) [inline src/geo/map/torque-layer.js]: Changed this in [09e10e742b714f7dc4fc7436522bb159f3c3961d](https://github.com/CartoDB/cartodb.js/pull/1268/commits/09e10e742b714f7dc4fc7436522bb159f3c3961d).

File: src/geo/map/torque-layer.js [hunk 2]
```diff
@@ -44,10 +49,18 @@ var TorqueLayer = LayerModelBase.extend({
   },
 
   _torqueTimeAttributeCartoCSSPropChanged: function () {
-    var regexp = /-torque-time-attribute:\s*["'](\w*)["']/;
-    var previousTorqueTimeAttributeProp = this.previous('cartocss').match(regexp) && this.previous('cartocss').match(regexp)[1];
-    var newTorqueTimeAttributeProp = this.get('cartocss').match(regexp) && this.get('cartocss').match(regexp)[1];
-    return newTorqueTimeAttributeProp !== previousTorqueTimeAttributeProp;
+    var currentCartoCSS = this.get('cartocss');
+    var previousCartoCSS = this.previous('cartocss');
+    var currentValue = this._getPropertyValueFromCartoCSS(currentCartoCSS, this.TORQUE_TIME_ATTRIBUTE_PROP);
```
Comment (javisantana) [inline src/geo/map/torque-layer.js]: you are parsing several times the cartocss and it's an expensive operation

Comment (javisantana) [inline src/geo/map/torque-layer.js]: you need to check not only time attribute, steps, resolution, column and aggregation are also needed

Comment (alonsogarciapablo) [inline src/geo/map/torque-layer.js]: > if you check more than one attrs you will be parsing

I don't understand what you mean by that...

I'm adding the rest of the attributes and I think I I'll need to parse both the new and old cartocss (so parsing will only be done twice)...

Comment (alonsogarciapablo) [inline src/geo/map/torque-layer.js]: What about `-torque-animation-duration` and `-torque-resolution:2` ?

Comment (javisantana) [inline src/geo/map/torque-layer.js]: -torque-animation-duration is managed client side, -torque-resolution needs a map reload (it's the pixel resolution, it changes the aggregation)

**Comment by javisantana** [comment]
tested on the editor, working fine :es:

**Comment by alonsogarciapablo** [comment]
Happy to hear that! 👍

**Pull request merged**